### PR TITLE
refactor: Average the atomic number in log

### PR DESCRIPTION
**Pull request by Corentin-Allaire**
Up until now the averaging of the material property (Ar and Z) were done using the number of mol per unit of surface of the materials. For Z this will lead to an incorrect computation of the energy loss which is proportional to `ln(Z)*thickness`.

This PR modify the way the value of Z is averaged (averaging the value of ln(Z) instead).

I do not know of any other use for Z than the computation of the mean excitation energy (and none are present in the acts files) but if there is another use for the value of Z then maybe we shouldn't change the way this is averaged (or change it differently).

### CODE REVIEW

**Review by asalzburger — DISMISSED** [review]

**Comment by asalzburger** [comment]
Isn't the thickness taken into account by `dE/dX * deltaT` - or is there a second dependency on the thickness ?

**Comment by Corentin-Allaire** [comment]
> Isn't the thickness taken into account by `dE/dX * deltaT` - or is there a second dependency on the thickness ?

I am not sure I fully understand the question, the thickness is indeed taken into account in `dE/dX * deltaT` with dE/dX ~ ln(Z)` so when you average your Z you need to average it as function of the thickness.
This function combine two material slap (lets call them 1 and 2) So if you imagine the situation where you go through 1 then two : 
`Eloss = El1+El2`
`Eloss ~ ln(Z1)*t1 + ln(Z2)*t2`
If you want to average Z so that : `Eloss ~ ln(Z)*t` with `t=t1+t2`
Then : `ln(Z)*t = ln(Z1)*t1 + ln(Z2)*t2`
Thus : `Z = exp( (t1/t)*ln(Z1) + (t2/t)*ln(Z2) )` which is the new formula we use to average Z with multiple material.
So there is no second dependency on the thickness but you average your two slab atomic number using their thickness.

**Review by asalzburger — APPROVED** [review]
Afer clarification this is now approved.
